zerver/lib/markdown/__init__.py

```diff
@@ -2100,6 +2115,13 @@ def possible_linked_stream_names(content: str) -> set[str]:
     }
 
 
+def possible_linked_topics(content: str) -> set[ChannelTopicInfo]:
+    return {
+        ChannelTopicInfo(match.group("stream_name"), match.group("topic_name"))
+        for match in re.finditer(STREAM_TOPIC_LINK_REGEX, content, re.VERBOSE)
+    }
```

Comment:
I think rather than parsing the message twice, we want to refactor `possible_linked_stream_names` to return more data from a single parse. 

I think `STREAM_TOPIC_MESSAGE_LINK_REGEX` is supposed to be accessed in `possible_linked_stream_names` too?

Reply:
I guess the existing code happens to work for the channel links because `STREAM_TOPIC_MESSAGE_LINK_REGEX` . And the topic code doesn't need this for message links, since those generate `/near/` URLs.